docs/bgp-policy.md

```diff
@@ -0,0 +1,223 @@
+
+- `address`: The IP address of the BGP peer.
+- `asn`: The Autonomous System Number of the BGP peer.
+- `port`: The port number on which the BGP peer listens. The default value is 179.
+- `multihopTTL`: The Time To Live (TTL) value used in BGP packets sent to the BGP peer, with a range of 1 to 255.
+  The default value is 1.
+- `gracefulRestartTimeSeconds`: Specifies how long the BGP peer waits for the BGP session to re-establish after a
+  restart before deleting stale routes, with a range of 1 to 3600 seconds. The default value is 120 seconds.
+
+## BGP router ID
+
+The BGP router identifier (ID) is a 4-byte field that is usually represented as an IPv4 address. Antrea uses the following
+steps to choose the BGP router ID:
+
+1. If the `node.antrea.io/bgp-router-id` annotation is present on the Node and its value is a valid IPv4 address string,
+   we will use the provided value.
+2. Otherwise, for an IPv4-only or dual-stack Kubernetes cluster, the Node's IPv4 address (assigned to the transport
+   interface) is used.
+3. Otherwise, for IPv6-only clusters, a 32-bit integer will be generated by hashing the Node name, then converted to the
+   string representation of an IPv4 address.
+
+After this selection process, the `node.antrea.io/bgp-router-id` annotation is added / updated as necessary to reflect
+the selected BGP router ID.
+
+## BGP Authentication
+
+BGP authentication ensures that BGP sessions are established and maintained only with legitimate peers. Users can provide
+authentication passwords for different BGP peering sessions by storing them in a Kubernetes Secret. The Secret must
+be defined in the same Namespace as Antrea (`kube-system` by default) and must be named `antrea-bgp-passwords`.
+
+By default, this Secret is not created, and BGP authentication is considered unconfigured for all BGP peers. If the
+Secret is created like in the following example, each entry should have a key that is the concatenated string of the BGP
+peer IP address and ASN (e.g., `192.168.77.100-65000`, `2001:db8::1-65000`), with the value being the password for that
+BGP peer. If a given BGP peer does not have a corresponding key in the Secret data, then authentication is considered
+disabled for that peer.
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: antrea-bgp-passwords
+  namespace: kube-system
+stringData:
+  192.168.77.100-65000: "password"
+  2001:db8::1-65000: "password"
+type: Opaque
+```
+
+## Example Usage
+
+### Combined Advertisements of Service, Pod, and Egress IPs
+
+In this example, we will advertise Service IPs of types ClusterIP and LoadBalancerIP, along with Pod CIDRs and Egress
+IPs from the selected Nodes to multiple remote BGP peers.
+
+```yaml
+apiVersion: crd.antrea.io/v1alpha1
+kind: BGPPolicy
+metadata:
+  name: advertise-all-ips
+spec:
+  nodeSelector:
+    matchLabels:
+      bgp: enabled
+  localASN: 64512
+  listenPort: 179
+  advertisements:
+    service:
+      ipTypes: [ClusterIP, LoadBalancerIP]
```

Comment:
I think we should remove ClusterIP to make the example more reasonable. Normally there is no reason to advertise ClusterIP when the cluster has LoadBalancer enabled.

Reply:
Done